src/game/spawn/player.rs

```diff
@@ -0,0 +1,36 @@
+//! Spawn the player.
+
+use bevy::prelude::*;
+
+use crate::{
+    game::physics::{PhysicalTransform, Velocity},
+    screen::Screen,
+};
+
+pub(super) fn plugin(app: &mut App) {
+    app.observe(spawn_player);
+}
+
+#[derive(Debug, Event)]
+pub(crate) struct SpawnPlayer;
+
+fn spawn_player(
+    _trigger: Trigger<SpawnPlayer>,
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+) {
+    commands.spawn((
+        Name::new("Player"),
+        SpriteBundle {
+            texture: asset_server.load("ducky.png"),
+            transform: Transform::from_scale(Vec3::splat(0.4)),
+            ..Default::default()
+        },
+        Velocity::default(),
+        // If your physics engine of choice uses `Transform` directly,
+        // a good hierarchy to follow instead is to have a `Player` root entity
+        // with the physical transform and the rendered transform as individual children.
+        PhysicalTransform::default(),
+        StateScoped(Screen::Playing),
```

Comment:
Ideally this would not be coupled. What if you want to like.. spawn a lil player entity in the title screen as a fun lil thing?

Fine to potentially change this in a later PR though.

Reply:
`Box<dyn Bundle>`, this will allow for state scoping as well as naming

Reply:
That might be fine. Hopefully bsn will take the wheel soon enough.

Reply:
> `Box<dyn Bundle>`, this will allow for state scoping as well as naming and any sort of marker components

Could you elaborate on that? I'm curious.

Reply:
`commands.spawn` uses `Commands::entities` to return an `EntityCommands` which includes both `Entity` and `Commands`.

`commands.add` and `commands.trigger` both can't do this, but an extension trait could (like we do for UI widgets). Maybe an extension trait with a generic method like `commands.my_spawn(impl Spawnable) -> EntityCommands`.

Reply:
Gonna resolve this conversation because it can be discussed further on discord and handled in a later PR.

Reply:
Leaving this for a follow-up